BUG: linprog failure - OptimizeResult.x returns NoneType

### Describe your issue.

NetworkX is seeing a regression in scipy-1.9.0rc1 in `scipy.optimize.linprog`. Comparing the `OptimizeResult.x` attribute to an integer threshold which worked in scipy 1.8 is now raising a `TypeError: '>=' not supported between instances of 'NoneType' and 'int'`.

See [this CI log](https://github.com/networkx/networkx/runs/7043690993?check_suite_focus=true) for details.

A quick look at `blame` suggests that the behavior change was likely introduced in the switch to `highs` as the default method #15512 .

### Reproducing Code Example

```python
Unfortunately the NX example is deeply embedded in a quite complicated test, so I don't have an MRE to hand. I will try to tease out a more minimal example than the above CI failure.
```


### Error message

```shell
Relevant snippet from above CI log::

            program_result = optimize.linprog(c, A_eq=a_eq, b_eq=b_eq)
>           bool_result = program_result.x >= 0
E           TypeError: '>=' not supported between instances of 'NoneType' and 'int'
```


### SciPy/NumPy/Python version information

1.9.0rc1 1.23.0 sys.version_info(major=3, minor=10, micro=4, releaselevel='final', serial=0)

--- a/scipy/optimize/_linprog.py
+++ b/scipy/optimize/_linprog.py
@@ -360,7 +360,7 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
         A :class:`scipy.optimize.OptimizeResult` consisting of the fields
         below. Note that the return types of the fields may depend on whether
         the optimization was successful, therefore it is recommended to check
-        `OptimizeResult.success` before relying on the other fields:
+        `OptimizeResult.status` before relying on the other fields:
 
         x : 1-D array
             The values of the decision variables that minimizes the

--- a/scipy/optimize/_linprog.py
+++ b/scipy/optimize/_linprog.py
@@ -357,7 +357,10 @@ def linprog(c, A_ub=None, b_ub=None, A_eq=None, b_eq=None,
     Returns
     -------
     res : OptimizeResult
-        A :class:`scipy.optimize.OptimizeResult` consisting of the fields:
+        A :class:`scipy.optimize.OptimizeResult` consisting of the fields
+        below. Note that the return types of the fields may depend on whether
+        the optimization was successful, therefore it is recommended to check
+        `OptimizeResult.success` before relying on the other fields:
 
         x : 1-D array
             The values of the decision variables that minimizes the

